### PR TITLE
Issue #4832: Get rid of SettingsActivity and add settings fragments to MainActivity.

### PR DESCRIPTION
**Pull request by pocmo**
This adds settings to the new navigation and turns Focus into an actual single activity app. This prevents the bug described in #3624 and hopefully makes further changes easier in the future.

Not all styles from the previous settings theme have made it over. I need to investigate this. Maybe there are visual changes in the redesign that can be combined with that in a new issue/pr.

### CODE REVIEW

**Comment by pocmo** [comment]
Looks like my last minute removal of code has caused a build error. I'll take a look :)

**Comment by pocmo** [comment]
Okay, now it's ready. :)

**Review by jonalmeida — APPROVED** [review]
Just a couple of questions, but looks great and ready to land! ✅

File: app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.kt [hunk 1]
```diff
@@ -53,12 +53,7 @@ class ManualAddSearchEngineSettingsFragment : BaseSettingsFragment() {
     override fun onResume() {
         super.onResume()
 
-        // We've checked that this cast is legal in super.onAttach.
```
[the remark this answers is not in the document]
Comment (pocmo) [reply app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.kt:56]: There are some fragments that we use in settings that are not a `PreferenceFragment` and just a regular `Fragment` and so I introduced a new base class I can use for those.

File: app/src/main/res/layout/fragment_autocomplete_add_domain.xml [hunk 2]
```diff
@@ -3,37 +3,54 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:orientation="vertical"
     android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    android:padding="8dp">
+    android:layout_height="match_parent">
 
-    <TextView
+    <View
```
[the remark this answers is not in the document]
Comment (pocmo) [reply app/src/main/res/layout/fragment_autocomplete_add_domain.xml:8]: Yeah, this view is behind the status bar, above the toolbar.

**Comment by jonalmeida** [comment]
I don't see any conflicts, so I'm going to land this.

**Pull request merged**